.github/workflows/test.yml

```
@@ -732,7 +732,9 @@ jobs:
             './gef/cypress/e2e/eligibility/**/*.spec.js',
             './gef/cypress/e2e/exporter/**/*.spec.js',
             './gef/cypress/e2e/external/**/*.spec.js',
-            './gef/cypress/e2e/facility/**/*.spec.js',
+            './gef/cypress/e2e/facility/*.spec.js',
```

Could you also split apart the non-ff tests too?

---

Also would it be more clear if there was a subdirectory for the remaining tests e.g 
```suggestion
            './gef/cypress/e2e/facility/pre-submission/**/*.spec.js',
```

Though some of the tests wouldn't match this description (e.g.review-ukef-approval-with-issued-and-unissued-facilities-MIA-and-resubmit-to-ukef.spec.js) & are not pre-submission - they could be moved to `unissued-facilities).

---

Thanks @MarRobSoftwire great shout, i've updated both E2E suites - agree some others could be split up, i've created the following directories to keep inline with the existing test names. It's structured like this now:

```
facility/*.spec.js
facility/end-date/*.spec.js
facility/review-ukef-approval/*.spec.js
facility/version-0/*.spec.js
facility/unissued-facilities/AIN/*.spec.js
facility/unissued-facilities/MIN/*.spec.js
```

What do you think?



Also, before merging this, want to monitor the time differences to see if we can make further improvements before merging.

---

Any `ff` specific E2E will need to be added to `e2e-tests-ff` job please, so we run all explicit truthy FF E2E coverage there and elsewhere it is standard system state.

---

This sounds great, what's left in `facility/*.spec.js`? Could they be moved into something like `'./gef/cypress/e2e/facility/pre-submission/**/*.spec.js'`?

Could `facility/end-date/*.spec.js` be renamed to `facility/facility-end-date/*.spec.js` & include the `bank-review-date` test? `facility/facility-end-date/*.spec.js` will not need to be tested when the ff is off

---

@abhi-markan It's already been added to `e2e-tests-ff`, it's hard to see with the git diff, need to expand previous lines 

https://github.com/UK-Export-Finance/dtfs2/pull/3482/commits/1056fb64084b8640fd24bdbdb5ba785692961ce6#diff-faff1af3d8ff408964a57b2e475f69a6b7c7b71c9978cccc8f471798caac2c88R714

---

Thanks @MarRobSoftwire 

I'm not keen on adding a `pre-submission` directory, I think it could be misleading/confusing because some existing tests/directories are "pre submission".

Have moved the `bank-review-date` test into the `end-date` directory, good shout

I think adding a `facility-` prefix to the `end-date` directory would be strange - in this case i'd expect all the directories to have a prefix and it'd become this which seems overkill because we have a `facility` parent directory

```js
facility/facility-end-date
facility/facility-review-ukef-approval
facility/facility-unissued-facilities
facility/facility-version-0
``` 

The existing structure/naming feels cleaner/simpler:

```js
facility/end-date
facility/review-ukef-approval
facility/unissued-facilities
facility/version-0
```

---

`facility-end-date` is a field the user submits (e.g. see https://ukef-dtfs.atlassian.net/browse/DTFS2-7220 ), like `bank-review-date` or `cover-start-date`. By calling it `end-date` this could be confused with `cover-end-date` (which is sometimes referred to as `endDate` in existing code). Moreover, if I'm searching for facility end date tests I wouldn't expect them to be under `end-date`

As to which tests are pre-submission, I think its only the facility end date ones so we could just put them in the same directory. Its more personal preference but I don't really like having some tests existing at a different level to others. I think it could encourage laziness to put a new facility test suite in the top directory, rather than checking if it could exist a level down.

Now I think about, the facility-end-date directory is maybe unnecessary and could be a point of confusion - `unissued-facilities-facility-end-date-AIN.spec.js` could live in the `unissued-facilities` or the `facility-end-date` directory. The tests currently in `facility-end-date` could just be moved into a pre-submission directory.

---

Thanks @MarRobSoftwire agreed, I have renamed `facility-end-date` to `pre-submission`.

I know what you mean about having tests in the parent directory, I had a think and I cannot think of a sensible way to group these tests (any ideas?), it seems they have nothing in common. In which case I think it makes sense to have them in a parent `facility` directory.

Sorry for any confusion, I am unfamiliar with theses tests, the intention of this PR is to split them up so that they run faster in GHA.